examples/deploy/terraform/cluster/variables.tf

```diff
@@ -1,7 +1,79 @@
-## Used to overwrite the `k8s_version` provided at initial creation.
-## When upgrading k8s, create/modify tfvars with desired `k8s_version` value.
-variable "k8s_version" {
-  description = "Update k8s version."
-  type        = string
-  default     = null
+## Used to overwrite the `eks` variable passed through the `infra` outputs.
+
+variable "eks" {
+  description = <<EOF
+    creation_role_name = Only meant to support an imported role.
+    k8s_version = EKS cluster k8s version.
+    kubeconfig = {
+      extra_args = Optional extra args when generating kubeconfig.
+      path       = Fully qualified path name to write the kubeconfig file.
+    }
+    public_access = {
+      enabled = Enable EKS API public endpoint.
+      cidrs   = List of CIDR ranges permitted for accessing the EKS public endpoint.
+    }
+    "Custom role maps for aws auth configmap
+    custom_role_maps = {
+      rolearn  = string
+      username = string
+      groups   = list(string)
+    }
+    master_role_names  = IAM role names to be added as masters in eks.
+    cluster_addons     = EKS cluster addons. vpc-cni is installed separately.
+    vpc_cni            = Configuration for AWS VPC CNI
+    ssm_log_group_name = CloudWatch log group to send the SSM session logs to.
+    identity_providers = Configuration for IDP(Identity Provider).
+  }
+  EOF
+
+  type = object({
+    creation_role_name = optional(string, null)
+    k8s_version        = optional(string)
+    kubeconfig = optional(object({
+      extra_args = optional(string)
+      path       = optional(string)
+    }), {})
+    public_access = optional(object({
+      enabled = optional(bool)
+      cidrs   = optional(list(string))
+    }), {})
+    custom_role_maps = optional(list(object({
+      rolearn  = string
+      username = string
+      groups   = list(string)
+    })))
+    master_role_names  = optional(list(string))
+    cluster_addons     = optional(list(string))
+    ssm_log_group_name = optional(string)
+    vpc_cni = optional(object({
+      prefix_delegation = optional(bool)
+    }))
+    identity_providers = optional(list(object({
+      client_id                     = string
+      groups_claim                  = optional(string)
+      groups_prefix                 = optional(string)
+      identity_provider_config_name = string
+      issuer_url                    = optional(string)
+      required_claims               = optional(string)
+      username_claim                = optional(string)
+      username_prefix               = optional(string)
+    })))
+  })
+
+  default = null
+}
+
+variable "kms_info" {
+  description = <<EOF
+    Overrides the KMS key information. Meant for migrated configurations.
```

Comment:
This is kind of like the earlier line, where it's not a functional part of the map, but the proceeding values are, so maybe we should put a `#` before this